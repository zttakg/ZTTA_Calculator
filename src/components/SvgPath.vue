<template>
  <path
    :d="path"
    :stroke-linecap="strokeLinecap"
    :stroke-width="strokeWidth"
    :stroke="strokeColor"
    :fill="fillColor"
  />
</template>

<script>
export default {
  name: "SvgPath",

  props: {
    path: {
      type: String,
      default: "M 100 300 L 150 350 L 100 400 L 50 350 L 100 300"
    },
    strokeWidth: Number,
    strokeLinecap: String,
    strokeColor: String,
    fillColor: String
  }
};
</script>
