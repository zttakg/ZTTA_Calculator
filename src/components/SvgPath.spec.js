import { shallowMount } from "@vue/test-utils";
import SvgPath from "./SvgPath";

describe("<SvgPath /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgPath);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(SvgPath, {
      propsData: {
        path: "M10 10 H 90 V 90 H 10 L 10 10",
        strokeWidth: 5,
        strokeLinecap: "butt",
        strokeColor: "#f06",
        fillColor: "red"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
