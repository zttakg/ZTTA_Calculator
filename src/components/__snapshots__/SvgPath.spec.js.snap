// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgPath /> component should render all props 1`] = `<path d="M10 10 H 90 V 90 H 10 L 10 10" stroke-linecap="butt" stroke-width="5" stroke="#f06" fill="red"></path>`;

exports[`<SvgPath /> component should render correct without props 1`] = `<path d="M 100 300 L 150 350 L 100 400 L 50 350 L 100 300"></path>`;
